### PR TITLE
Improve api validation unit test coverage

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

Add UT for ValidatePodLogOptions and AccumulateUniqueHostPorts

Before

```
coverage: 60.0% of statements
ok  	k8s.io/kubernetes/pkg/api/v1/validation	0.058s
```

After:
```
coverage: 92.9% of statements
ok  	k8s.io/kubernetes/pkg/api/v1/validation	0.057s
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by thockin** [comment]
@yujuhong because she knows log APIs semantics

File: pkg/apis/core/v1/validation/validation_test.go [hunk 1]
```diff
@@ -177,3 +179,226 @@ func TestValidateResourceRequirements(t *testing.T) {
...
+				},
+			},
+			accumulator: &sets.String{},
+			fldPath:     field.NewPath("spec", "containers"),
+			result:      "HostPort is not allocated",
+		},
+	}
+	for _, tc := range successCase {
+		if errs := AccumulateUniqueHostPorts(tc.containers, tc.accumulator, tc.fldPath); len(errs) != 0 {
+			t.Errorf("unexpected error: %v", errs)
```
Comment (yujuhong) [inline pkg/apis/core/v1/validation/validation_test.go]: Should print out which test case this error belongs to.

File: pkg/apis/core/v1/validation/validation_test.go [hunk 2]
```diff
@@ -177,3 +179,226 @@ func TestValidateResourceRequirements(t *testing.T) {
...
+				},
+			},
+			accumulator: &sets.String{"8080/UDP": sets.Empty{}},
+			fldPath:     field.NewPath("spec", "containers"),
+			result:      "HostPort is already allocated",
+		},
+	}
+	for _, tc := range errorCase {
+		if errs := AccumulateUniqueHostPorts(tc.containers, tc.accumulator, tc.fldPath); len(errs) == 0 {
+			t.Errorf("expected error %v, but get nil", tc.result)
```
Comment (yujuhong) [inline pkg/apis/core/v1/validation/validation_test.go]: Pame here: print out which test case this error belongs to.

**Comment by wackxu** [comment]
@yujuhong Updated, PTAL

**Pull request merged**